src/datadog/filter/matcher.rs

```diff
@@ -99,6 +99,7 @@ where
 /// # Errors
 ///
 /// Will return `Err` if the query contains an invalid path.
+#[allow(clippy::module_name_repetitions)]
 pub fn build_matcher<V, F>(
```

Comment:
I kinda agree the method name is redundant here. We could just call this `build`.

Reply:
Ah, good point. That might be worth leaving a comment to indicate why we are ignoring the clippy lint.

If we wanted to, we could go further to introduce the new methods with new names and deprecate these ones.